File: app/src/main/java/org/deiverbum/app/model/LHIntercession.java
package org.deiverbum.app.model;

import static org.deiverbum.app.utils.Constants.TITLE_INTERCESSIONS;
import static org.deiverbum.app.utils.Utils.LS2;

import android.text.SpannableStringBuilder;
import android.text.Spanned;

import org.deiverbum.app.utils.Utils;

import java.util.Locale;

public class LHIntercession {
    public Integer intercessionID;
    public String intro;
    public String intercession;

    public String getIntro() {
        return intro;
    }

    public void setIntro(String intro) {
        this.intro = intro;
    }

    public SpannableStringBuilder getAll() {

        SpannableStringBuilder sb=new SpannableStringBuilder();
        sb.append(getHeader());
        sb.append(LS2);
        String[] introArray = intro.split("\\|");
        if (introArray.length == 3) {
            sb.append(introArray[0]);
            sb.append(LS2);
            sb.append(Utils.fromHtml(String.format(new Locale("es"),"<i>%s</i>",introArray[1])));
            sb.append(LS2);
            sb.append(getIntercession());
            sb.append(LS2);
            sb.append(introArray[2]);
        } else {
            sb.append(getIntro());
            sb.append(LS2);
            sb.append(getIntercession());
        }
        return sb;

    }

    public Spanned getIntercession() {
        return Utils.fromHtml(Utils.getFormato(intercession));
    }

    public void setIntercession(String intercession) {
        this.intercession = intercession;
    }

    public SpannableStringBuilder getHeader() {
        return Utils.formatTitle(TITLE_INTERCESSIONS);
    }

    public String getHeaderForRead() {
        return Utils.pointAtEnd(TITLE_INTERCESSIONS);
    }

    public SpannableStringBuilder getAllForRead() {
        SpannableStringBuilder sb=new SpannableStringBuilder();
        sb.append(getHeaderForRead());
        sb.append(LS2);
        String[] introArray = intro.split("\\|");
        if (introArray.length == 3) {
            sb.append(introArray[0]);
            sb.append(introArray[1]);
            sb.append(getIntercession());
            sb.append(introArray[2]);
        } else {
            sb.append(getIntro());
            sb.append(getIntercession());
        }
        return sb;
    }
}

File: app/src/main/java/org/deiverbum/app/model/LHIntercession.kt
package org.deiverbum.app.model

import android.text.SpannableStringBuilder
import android.text.Spanned
import org.deiverbum.app.util.Constants
import org.deiverbum.app.util.Utils
import java.util.*

class LHIntercession {
    var intercessionID: Int? = null
    var intro: String? = null
    var intercession: String? = null
    val all: SpannableStringBuilder
        get() {
            val sb = SpannableStringBuilder()
            sb.append(header)
            sb.append(Utils.LS2)
            val introArray =
                intro!!.split("\\|".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            if (introArray.size == 3) {
                sb.append(introArray[0])
                sb.append(Utils.LS2)
                sb.append(Utils.fromHtml(String.format(Locale("es"), "<i>%s</i>", introArray[1])))
                sb.append(Utils.LS2)
                sb.append(getIntercessionForView())
                sb.append(Utils.LS2)
                sb.append(introArray[2])
            } else {
                sb.append(intro)
                sb.append(Utils.LS2)
                sb.append(getIntercessionForView())
            }
            return sb
        }

    private fun getIntercessionForView(): Spanned {
        return Utils.fromHtml(Utils.getFormato(intercession))
    }


    val header: SpannableStringBuilder
        get() = Utils.formatTitle(Constants.TITLE_INTERCESSIONS)
    private val headerForRead: String
        get() = Utils.pointAtEnd(Constants.TITLE_INTERCESSIONS)
    val allForRead: SpannableStringBuilder
        get() {
            val sb = SpannableStringBuilder()
            sb.append(headerForRead)
            sb.append(Utils.LS2)
            val introArray =
                intro!!.split("\\|".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            if (introArray.size == 3) {
                sb.append(introArray[0])
                sb.append(introArray[1])
                sb.append(getIntercessionForView())
                sb.append(introArray[2])
            } else {
                sb.append(intro)
                sb.append(getIntercessionForView())
            }
            return sb
        }
}